### PR TITLE
EnemyHandler

### PR DESCRIPTION
**Pull request by OtakuFreak13**
Made an enemy handler in Game that makes it easy to add enemies to a scene.
The Handler saves the information of an enemy type to a struct the first time that type is added to a scene. This struct is then used to make the enemy the following times with only enemy name and position needed as input.

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/EnemyHandler.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	// to skipp scale and write to rot the equvalent would be FLT_MAX
+	// OBS "pos" goes BEFORE "entityName" in this overloaded function
+	Entity* AddEnemy(float3 pos, std::string entityName, unsigned int flag = UINT_MAX, float scale = FLT_MAX, float3 rot = { FLT_MAX , FLT_MAX , FLT_MAX });
+
+
+
+
+private:
+	Scene* m_pScene = nullptr;
+	std::map<std::string, EnemyComps*> m_enemyComps;
```
Comment (Whixx) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyHandler.h]: Stor bokstav i m_enemyComps

**Review by Whixx — CHANGES_REQUESTED** [review]
Look at using more clear names for variables and classes which describe what the actually do. Mainly i want you to think about the name "EnemyHandler", which functions of the enemies do this class actually handle? F;r example if the task of the class is just to produce new enemies i'd say "EnemyFactory" or something similar is more fitting.
Also have another look through the code for codestandard reasons. I've made a comment on a member variable that is not correct.

File: StortSpelProjekt/Game/src/Gamefiles/EnemyHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+				Physics::GetInstance().AddCollisionEntity(ent);
+			}
+			return ent;
+		}
+		else
+		{
+			Log::PrintSeverity(Log::Severity::WARNING, "Inssuficient input in parameters to add new type of enemy!\n");
+			return nullptr;
+		}
+	}
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyHandler.cpp]: The naming of the three overloaded functions are a bit confusing, change according to what we discussed in Discord.

File: StortSpelProjekt/Game/src/Gamefiles/EnemyHandler.h (hunk 1 above)
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyHandler.h]: m_EnemyComps

File: StortSpelProjekt/Game/src/main.cpp [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #include "Engine.h"
 #include "Components/PlayerInputComponent.h"
+#include "EnemyHandler.h"
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/main.cpp]: Should this be included here? This might be the case, but maybe it should be included somewhere in "game" ?

**Review by jocke1995 — CHANGES_REQUESTED** [review]
Change according to my comments

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.h [hunk 4]
```diff
@@ -0,0 +1,57 @@
+#ifndef ENEMY_HANDLER_H
+#define ENEMY_HANDLER_H
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.h]: Should be ENEMYFACTORY_H

**Review by jocke1995 — CHANGES_REQUESTED** [review]
Just a small naming bug from when you changed the name of the class.

**Review by Whixx — APPROVED** [review]

**Review by jocke1995 — APPROVED** [review]

**Pull request merged**